### PR TITLE
fix(Form): Fix Form.Item not rendering correctly in Modal when it has help

### LINKED ISSUE #40500: Modal中的Form.Item设置了help后，重置表单导致原Form.Item的marginBottom发生变化
### Reproduction link

[https://github.com/xihayouyi/demo1.git](https://github.com/xihayouyi/demo1.git)
[http://oss.cn-software.top/](http://oss.cn-software.top/)

### Steps to reproduce

1）点击表单的”重置“按钮
2）设置了"help"的表单项的marginBottom发生了变化

### What is expected?

点击表单的”重置“按钮后，显式的设置了'help'的表单项的marginBottom不应该有变化

### What is actually happening?

点击了表单的”重置“按钮后，设置了'help'的表单项的marginBottom间距变化了

| Environment | Info |
| --- | --- |
| antd | undefined |
| React | 18.2.0 |
| System | macOS Monterey12.5 |
| Browser | Chrome108.0.5359.98 |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/40500

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix `Form.Item` not rendering correctly in `Modal` when it has `help`           |
| 🇨🇳 Chinese | 修复 `Form.Item` 存在 help 值时在 `Modal` 中没有正确渲染元素         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by MadCcc** [comment]
Ref change will not trigger rerender, and this effect may not work.
Any other idea?

**Comment by Yuiai01** [comment]
> Ref change will not trigger rerender, and this effect may not work. Any other idea?

Ok, let me think again.

**Comment by Yuiai01** [comment]
@MadCcc This problem is because the `itemRef` is not rendered at the first time, resulting in null values obtained, and `useLayoutEffect` can get the latest dom before the dom is actually updated. The way I think of now is to use `useEffect` in addition to listening to`itemRef`. o(╥﹏╥)o

File: components/form/FormItem/ItemHolder.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export default function ItemHolder(props: ItemHolderProps) {
       const itemStyle = getComputedStyle(itemRef.current);
       setMarginBottom(parseInt(itemStyle.marginBottom, 10));
     }
-  }, [hasError]);
+  }, [hasError, marginBottom]);
```
Comment (MadCcc) [inline components/form/FormItem/ItemHolder.tsx:75]: `marginBottom` 改变的时候 setMarginBottom，逻辑看上去怪怪的，是不是应该 `help` 在 deps 里？

Comment (Yuiai01) [reply components/form/FormItem/ItemHolder.tsx:75]: > `marginBottom` 改变的时候 setMarginBottom，逻辑看上去怪怪的，是不是应该 `help` 在 deps 里？

`hasError ` 的值是由 `help` 推导来的，放在 deps 里和单独 `hasError ` 没太大区别，这个问题原因是因为itemStyle.marginBottom 一直没有获取到正确的值，所以将其添加至 deps 。

File: components/form/FormItem/ItemHolder.tsx [hunk 2]
```diff
@@ -65,14 +65,18 @@ export default function ItemHolder(props: ItemHolderProps) {
...
-      setMarginBottom(parseInt(itemStyle.marginBottom, 10));
+    if (hasError && itemRef.current && !marginBottom) {
+      const computedStyle = getComputedStyle(itemRef.current);
+      if (computedStyle) {
+        setItemStyle(computedStyle);
+        setMarginBottom(parseInt(computedStyle.marginBottom, 10));
+      }
     }
-  }, [hasError]);
+  }, [hasError, itemStyle]);
```
Comment (zombieJ) [inline components/form/FormItem/ItemHolder.tsx:79]: 在 effect 里 setState 就不推荐在 deps 里也作为条件了，因为这样非常容易死循环（尤其是 `getComputedStyle` 这个返回的总是一个新的对象）。如果是需要按照 `marginBottom` 变化触发的话，可以调整一下，把 `marginBottom` 放到条件里，并且当 computedStyle 没有的时候给 `marginBottom` 设置 `null` 来做到状态切换。

Comment (Yuiai01) [reply components/form/FormItem/ItemHolder.tsx:79]: > 在 effect 里 setState 就不推荐在 deps 里也作为条件了，因为这样非常容易死循环（尤其是 `getComputedStyle` 这个返回的总是一个新的对象）。如果是需要按照 `marginBottom` 变化触发的话，可以调整一下，把 `marginBottom` 放到条件里，并且当 computedStyle 没有的时候给 `marginBottom` 设置 `null` 来做到状态切换。

好的～

**Review by zombieJ — APPROVED** [review]

**Pull request merged**